platform/wayland/cog-platform-wl.c

```diff
@@ -980,11 +1029,21 @@ touch_on_down(void              *data,
               wl_fixed_t         x,
               wl_fixed_t         y)
 {
-    CogWlPlatform *platform = data;
-    CogWlDisplay  *display = platform->display;
+    CogWlSeat    *seat = data;
+    CogWlDisplay *display = seat->display;
 
-    display->seat.touch.surface = surface;
-    display->seat.keyboard.serial = serial;
+    if (touch != seat->touch_obj) {
+        g_critical("%s: Got touch %p, expected %p.", G_STRFUNC, touch, seat->touch_obj);
+        return;
+    }
+
+    seat->display->seat_default = seat;
+
+    CogWlPlatform *platform =
+        wl_surface_get_user_data(surface); // Platform is attached the context of the wl_surface. Check create_window();
+    seat->touch_target = platform;
```

Comment:
Associating the Platform object as the context of the touch event (this is a pattern that also happens in the pointer_on_enter and in the keyboard_on_key too). The idea is these events, who they dont get the reference to the wl_surface directly, they gets where this event is being happening thanks to this info in the context. Today they only get a reference to the Platform what is so obvious but later this could be different references, for example, if the View can be contained by difference instances (ViewPort concept, Group of views using different Windows?)